runtime/router/trie.go

```diff
@@ -235,21 +245,21 @@ func (t *tnode) set(path string, value http.Handler, lastKeyCharSlash, lastPathC
 				key:   path[i:],
 				value: value,
 			}
-			t.children = append(t.children, newNode)
+			t.addChildren(newNode, lastPathCharSlash)
 		}
 	}
 
 	return nil
 }
 
-func (t *tnode) get(path string, lastKeyCharSlash, lastPathCharSlash, colonAsPattern bool) (http.Handler, []Param, error) {
+func (t *tnode) get(path string, lastKeyCharSlash, lastPathCharSlash, colonAsPattern, isWhitelistedPath bool) (http.Handler, []Param, error) {
 	keyLength, pathLength := len(t.key), len(path)
 	var params []Param
 
 	// find the longest matched prefix
 	var keyIdx, pathIdx int
 	for keyIdx < keyLength && pathIdx < pathLength {
-		if t.key[keyIdx] == ':' && lastKeyCharSlash {
+		if t.isWildCardPattern(path, keyIdx, pathIdx, lastKeyCharSlash, lastPathCharSlash, colonAsPattern, isWhitelistedPath) {
```

Comment:
t.key[keyIdx] == ':' && lastKeyCharSlash && colonAsPattern is sufficient here

Reply:
It is not sufficient and it will not work for set (set invokes this get func) for the below sequence for non-whitelisted cases
/a/b
/a/:var
Here we expect the second path to fail but with your logic it will succeed.

Reply:
sure, but can we make a specific function for getter and setter so we know which path is needed and which is redundant. makes it easier to maintain.

Reply:
The logic is same for both and it is controlled by colonAsPattern flag. This flag is not something that I introduced.

Reply:
Created separate functions for getter and setter